### ISSUE TITLE
Detailed output not printing when multiple gridlines specify a random seed

### ISSUE THREAD

**Issue by reinhold-willcox** [TeamCOMPAS/COMPAS#548]
**Describe the bug**
This is a pretty specific corner case, but I think I've reduced this to a MWE.

Detailed output is not working correctly, precisely when a gridfile is used which has the --random-seed parameter specified *more than once*. 

If detailed output is turned off, or if the gridfile specifies --random-seed on only one line (or none), the output seems to work fine. However, if there is more than one line with --random-seed specified, the following error is output:

```
ERROR: Unable to create HDF5 detailed file with file name /home/rwillcox/astro/compas/COMPAS/src/COMPAS_Output_7/Detailed_Output/BSE_Detailed_Output_0.h5
Logging disabled
Error writing to file - data not written while writing to logfile BSE_System_Parameters
Segmentation fault (core dumped)
```

This occurs on the current most up to date `dev` branch, as well as quite a few past ones (all the ones I tested, though at some point far enough back the code/tests don't work anymore). It also occurred for me on my desktop (Ubuntu 18.04) and on OzStar, so I don't think this is an OS issue.

**Label the issue**
Please label the 'severity' and 'urgency' of this issue. You can choose:

`urgency_low`      - This issue is not urgent
`severity_minor`    - This is a minor bug with minimal impact

**To Reproduce**
Create grid.txt:
```
--random-seed 0 --initial-mass-1 10    
--random-seed 1 --initial-mass-1 10   
```
And run `COMPAS --detailed --grid grid.txt`. You should see the error. Then edit grid.txt:
```
--random-seed 0 --initial-mass-1 10    
--initial-mass-1 10   
```
and rerun, there should be no error.

**Expected behavior**
The detailed output should be produced for all the systems, regardless of the random-seed. 

**Versioning (please complete the following information):**
 - Ubuntu 18.04
 - COMPAS v02.19.00

**Comment by jeffriley** [comment]
Good catch Reinhold!  In fact, this has been a problem since the new grid files were implemented, and (a variation of) it does happen for other logfile types (a less benign variation).

e.g. running a test with the following grid file:

--eccentricity 0.11 --random-seed 12345
--eccentricity 0.12 --random-seed 23456
--eccentricity 0.13 --random-seed 34567

and commandline:

./compas --grid grid.txt --detailed-output --logfile-type csv

results in detailed output CSV files being created, but their names are malformed.  The following detailed output files will be created:

BSE_Detailed_Output_0.csv
BSE_Detailed_Output_0_1.csv
BSE_Detailed_Output_0_2.csv

The files that *should* be created are:

BSE_Detailed_Output_0.csv
BSE_Detailed_Output_1.csv
BSE_Detailed_Output_2.csv

The reason creating the files doesn't fail for CSV (and TSV and TXT) files is that the COMPAS code checks if the file already exists, then adds an incremental suffix if it does - in the case above COMPAS is (erroneously) trying to create "BSE_Detailed_Output_0" for each binary, and after the first adds the suffix "_1", then "_2".  That's incorrect behaviour (adding the suffix is correct - but being in that situation is not), but it has gone unnoticed until now.

Logging to HDF5 files is a little different from the other logfile formats - there is no versioning of a filename that already exists, so if the HDF5 already exists the code just reports an error and logging fails.  Maybe I should add versioning - I'll look into that.

The real problem is that the suffix for the detailed output file ("_0", "_1", etc.) isn't being generated correctly when a grid file contains the random-seed option.  I know how to fix it - I will push a fix in the next couple of days if I don't get it done today.

**Comment by reinhold-willcox** [comment]
Thanks Jeff, glad it's an easy fix!

**Comment by jeffriley** [comment]
Introduced in v02.16.00
Fixed in v02.19.01 (PR #550 )
